### PR TITLE
tools/horizon-cmp: Retry on Latest-Ledger header mismatch

### LINKED ISSUE #2017: tools/horizon-cmp: Use Latest-Ledger header
In #1791 we added `Latest-Ledger` header. We should check it in `horizon-cmp` and repeat requests later if `Latest-Ledger` is different in base and test servers. This will decrease the number of false positives.

### PR DESCRIPTION
**Pull request by bartekn**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

This commit queues a path to retry later when `Latest-Ledger` header is not equal.

### Why

If `Latest-Ledger` is not equal horizon-cmp can report the given path even though it's fine if responses are not equal.

Close #2017.

### CODE REVIEW

File: tools/horizon-cmp/main.go [hunk 1]
```diff
@@ -183,6 +183,16 @@ func run(cmd *cobra.Command) {
 
 			requestWg.Wait()
 
+			// Retry when LatestLedger not equal but only if not empty because
+			// older Horizon versions don't send this header.
+			if a.LatestLedger != "" && b.LatestLedger != "" &&
+				a.LatestLedger != b.LatestLedger {
+				visitedPaths[pl.ID()] = false
```
Comment (tamirms) [inline tools/horizon-cmp/main.go]: is it safe to update `visitedPaths` from multiple goroutines?

**Comment by tamirms** [comment]
I wonder if closing the paths channel is safe in `addPathsFromResponse()`
https://github.com/stellar/go/blob/3392e7b5a46f89bfef2376b47bfafd06eaacb538/tools/horizon-cmp/main.go#L345-L347

It seems like it could be possible that paths is empty but there could be multiple go routines that are still comparing horizon responses. The other go routines might attempt to add to the paths channel after it has been closed.

File: tools/horizon-cmp/main.go (hunk 1 above)
Comment (bartekn) [reply tools/horizon-cmp/main.go]: Good point, fixed in https://github.com/stellar/go/pull/2286/commits/ea2af4eb91c1a67063f52da9c31f2d03248a4ba0.

**Comment by bartekn** [comment]
> It seems like it could be possible that paths is empty but there could be multiple go routines that are still comparing horizon responses. The other go routines might attempt to add to the paths channel after it has been closed.

You're right! Going to create an issue for this.

**Pull request merged**